pkg/roachprod/roachprod.go

```diff
@@ -1450,7 +1461,16 @@ func Destroy(
 				// and let the caller retry.
 				cld, _ = cloud.ListCloud(l, vm.ListOptions{IncludeEmptyClusters: true})
 			}
-			return destroyCluster(ctx, cld, l, name)
+			err := destroyCluster(ctx, cld, l, name)
+			if err != nil {
+				return errors.Wrapf(err, "unable to destroy cluster %s", name)
```

Comment:
I wonder if this could lead to a split-brain scenario? Say, `destroyCluster` actually destroys a cluster but fails for some other reason. In that case, the local state is still there.

Reply:
Local state would be updated during next `List` or `Sync`.

This is, btw, the current behavior as today the local cache is never deleted when a cluster is destroyed.